cranelift/codegen/src/loop_analysis.rs

```diff
@@ -29,6 +30,39 @@ pub struct LoopAnalysis {
 struct LoopData {
     header: Block,
     parent: PackedOption<Loop>,
+    level: LoopLevel,
+}
+
+/// A level in a loop nest.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct LoopLevel(u8);
+impl LoopLevel {
+    /// Get the root level (no loop).
+    pub fn root() -> Self {
+        Self(0)
+    }
+    /// Get the loop level.
+    pub fn level(self) -> usize {
+        self.0 as usize
+    }
+    /// Invalid loop level.
+    pub fn invalid() -> Self {
+        Self(0x80)
```

Comment:
I'm curious, why pick 0x80 as the invalid level, instead of 0xFF?

Reply:
To be honest, I have no idea -- lost to the mists of time! I've reworked this to use a sentinel of `0xff` and hopefully be a little clearer.